tensorflow_quantum/core/qsim/state_space_test.cc

```diff
@@ -18,15 +18,380 @@ limitations under the License.
+  auto state = std::unique_ptr<StateSpace>(GetStateSpace(2, 1));
+  state->CreateState();
+
+  std::complex<float> ampl_00(0.1, 0.5);
+  std::complex<float> ampl_01(0.2, 0.6);
+  std::complex<float> ampl_10(0.3, 0.7);
+  std::complex<float> ampl_11(0.4, 0.8);
+
+  state->SetAmpl(0, ampl_00);
+  state->SetAmpl(1, ampl_01);
+  state->SetAmpl(2, ampl_10);
+  state->SetAmpl(3, ampl_11);
+
+  ASSERT_EQ(state->GetAmpl(0), ampl_00);
+  ASSERT_EQ(state->GetAmpl(1), ampl_01);
+  ASSERT_EQ(state->GetAmpl(2), ampl_10);
+  ASSERT_EQ(state->GetAmpl(3), ampl_11);
+
+  state->SetStateZero();
+  ASSERT_EQ(state->GetAmpl(0), std::complex<float>(1, 0));
+  ASSERT_EQ(state->GetAmpl(1), std::complex<float>(0, 0));
+  ASSERT_EQ(state->GetAmpl(2), std::complex<float>(0, 0));
+  ASSERT_EQ(state->GetAmpl(3), std::complex<float>(0, 0));
+}
+
+TEST(StateSpaceTest, CopyFromGetRealInnerProduct) {
+  auto state = std::unique_ptr<StateSpace>(GetStateSpace(12, 1));
+  state->CreateState();
+  for (uint64_t i = 0; i < state->GetDimension(); i++) {
+    state->SetAmpl(i, std::complex<float>(i, i));
+  }
+
+  auto state_clone = std::unique_ptr<StateSpace>(state->Clone());
+  state_clone->CreateState();
+  state_clone->CopyFrom(*state);
+
+  for (uint64_t i = 0; i < state->GetDimension(); i++) {
+    ASSERT_EQ(state->GetAmpl(i), state_clone->GetAmpl(i));
+  }
+
+  double real_inner_product = 0.0;
+  for (uint64_t i = 0; i < state->GetDimension(); i++) {
+    real_inner_product += 2 * i * i;
+  }
+
+  EXPECT_NEAR(state->GetRealInnerProduct(*state_clone), real_inner_product,
+              1E-2);
+}
+
+TEST(StateSpaceTest, ApplyGate1) {
+  auto state = std::unique_ptr<StateSpace>(GetStateSpace(5, 1));
+  state->CreateState();
+  const float matrix_h[] = {1.0 / std::sqrt(2), 0.0, 1.0 / std::sqrt(2),  0.0,
+                            1.0 / std::sqrt(2), 0.0, -1.0 / std::sqrt(2), 0.0};
+  state->SetStateZero();
+  state->SetAmpl(0, std::complex<float>(0.0, 0.0));
+  state->SetAmpl(1, std::complex<float>(1.0, 0.0));
+  switch (state->GetType()) {
+    case StateSpaceType::AVX:
+      ASSERT_EQ(
+          state->ApplyGate1(matrix_h),
+          tensorflow::Status(tensorflow::error::INVALID_ARGUMENT,
+                             "AVX simulator doesn't support small circuits."));
+      break;
+    case StateSpaceType::SLOW:
+      state->ApplyGate1(matrix_h);
+      ASSERT_EQ(state->GetAmpl(0), std::complex<float>(1 / std::sqrt(2), 0.0));
+      ASSERT_EQ(state->GetAmpl(1), std::complex<float>(-1 / std::sqrt(2), 0.0));
+      break;
+    case StateSpaceType::SSE:
+      ASSERT_EQ(
+          state->ApplyGate1(matrix_h),
+          tensorflow::Status(tensorflow::error::INVALID_ARGUMENT,
+                             "SSE simulator doesn't support small circuits."));
+      break;
+  }
+}
+
+TEST(StateSpaceTest, ApplyGate2) {
+  auto state = std::unique_ptr<StateSpace>(GetStateSpace(6, 1));
+  state->CreateState();
+  const float matrix_hi[] = {1.0 / std::sqrt(2),
```

Comment:
Can this formatting be more human readable ?

Reply:
Reverted and added clang format guards